### PR TITLE
Clarify O3DE CLI examples

### LINKED ISSUE #806: [ISSUE][DOCS]CLI Reference `create-template` | Add more usage examples
**This issue is blocked until PR https://github.com/o3de/o3de.org/pull/763 is merged, which will create the O3DE CLI Reference page.** 

## Describe the issue briefly

This issue is to update the O3DE CLI Reference page. 

We need more usage examples on how to use the `create-template` command in the `o3de` Python script. This command is complex and has many parameters, and there are many ways users can create templates. We need additional usage examples that cover these different use cases. 

## Which page(s) / section(s) are affected?

**File**: content\docs\user-guide\project-config\cli-reference.md
**Website**: https://o3de.org/docs/user-guide/project-config/cli-reference/#usage-2
_(If the anchor link doesn't work, go to the section: `create-template` > Usage .)_

## Does this work have an engineering dependency? What is it?

@lumberyard-employee-dm and sig-build for more info on this command and usage examples.

### PR DESCRIPTION
**Pull request by willihay**
Signed-off-by: William Hayward <wilhayw@amazon.com>

## Change summary

- Adds usage examples and clarifications for several O3DE CLI commands.
- Adds a recommended usage of `-kr` parameter ("keep restricted") when creating components from the default component template. This, oddly, _avoids_ the creation and registration of a "restricted" folder in the `<user>/.o3de/Restricted/Templates` directory when instantiating from this template.
- Adds a usage note for Windows PowerShell users who use the o3de script with the `--replace` parameter.

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

Fix #806

### CODE REVIEW

**Review by micronAMZN — APPROVED** [review]

**Review by chanmosq — APPROVED** [review]
Thanks for troubleshooting this with me!

File: content/docs/user-guide/project-config/cli-reference.md [hunk 1]
```diff
@@ -268,19 +301,27 @@ o3de.bat create-template --source-path SOURCE_PATH --template-path TEMPLATE_PATH
 
 - **`-kr, --keep-restricted-in-template`**
 
-  If true, creates the restricted platforms in the template folder. If false, creates the restricted files in the restricted folder located at TEMPLATE_RESTRICTED_PATH. By default, this parameter is false.
+  If included, creates the restricted platforms in the template folder. If false, creates the restricted files in the restricted folder located at TEMPLATE_RESTRICTED_PATH. By default, this parameter is false.
```
Comment (chanmosq) [inline content/docs/user-guide/project-config/cli-reference.md:304]:
```suggestion
  If included, creates the restricted platforms in the template folder. By default, creates the restricted files in the restricted folder located at TEMPLATE_RESTRICTED_PATH.
```

File: content/docs/user-guide/project-config/cli-reference.md [hunk 2]
```diff
@@ -268,19 +301,27 @@ o3de.bat create-template --source-path SOURCE_PATH --template-path TEMPLATE_PATH
...
 
 - **`-kl, --keep-license-text`**
 
-  If true, keeps the license text (located in the `template.json` file) in the new template's `template.json` file. If false, the license text isn't included. By default, this parameter is false. The license text is all of the lines of text, starting at {BEGIN_LICENSE} and ending at {END_LICENSE}.
+  If included, keeps all of the lines of license text, starting at {BEGIN_LICENSE} and ending at {END_LICENSE}. If false, the license text isn't included.
```
Comment (chanmosq) [inline content/docs/user-guide/project-config/cli-reference.md:308]:
```suggestion
  If included, keeps all of the lines of license text, starting at {BEGIN_LICENSE} and ending at {END_LICENSE}. By default, the license text isn't included.
```

Comment (chanmosq) [inline content/docs/user-guide/project-config/cli-reference.md:308]:
```suggestion
  If included, keeps all of the lines of license text, starting at {BEGIN_LICENSE} and ending at {END_LICENSE}. By default, the license text isn't included.
```

**Review by chanmosq — APPROVED** [review]

**Review by micronAMZN — APPROVED** [review]

**Pull request merged**